list/List.js

```diff
@@ -132,24 +133,47 @@ define([
 		// the SPACE key to (de)select an item.
 		multiCharSearchDuration: 0,
 
-		setAttribute: dcl.superCall(function (sup) {
-			return function (attr, value) {
-				sup.apply(this, arguments);
-				if (attr === "role") {
-					// Only role=grid should have aria-readonly.  It's meaningless for role=listbox or role=menu.
-					if (value === "grid") {
-						this.setAttribute("aria-readonly", "true");
-					} else {
-						this.removeAttribute("aria-readonly");
-					}
+		/**
+		 *	If `true` and the list's store is empty, the NoItemContainer element will be shown.
+		 * @member {boolean}
+		 * @default "false"
+		 */
+		showNoItems: false,
 
-					// Update roles of existing renderers.
-					this.getRenderers().forEach(function (renderer) {
-						renderer.parentRole = value;
-					});
-				}
-			};
-		}),
+		/**
+		 * Message to display when the d-list-no-items node is shown.
+		 * @member {string}
+		 * @default "Nothing to show."
+		 */
+		noItemsInfo: messages["no-items-info"],
+
+		/**
+		 * Flag set to a truthy value once the items initialization starts.
+		 * It is set to false when the initialization ends.
+		 * Furthermore if `true`, a loading panel will be shown.
+		 * @member {boolean}
+		 * @private
+		 */
+		_busy: false,
+
+		/**
+		 * Specifies the role of list. It can be one of `grid`, `menu`, `listbox`.
+		 * @type {String}
+		 * @default "grid"
+		 */
+		type: "grid",
+
+		/**
+		 * Propertie responsible for showing/hiding the containerNode element, depending of the
+		 * content of the source. It also drives the visibility of d-list-no-items node,
+		 * accordingly to the showNoItems's value.
+		 * @type {String}
+		 * @default ""
+		 * @private
+		 */
+		_displayedPanel: "",
```

Comment:
Can you list the possible values here?  Also, the description is a bit weird; it says that it's responsible for "showing/hiding the containerNode" but that's only one of the things it does.